### DISCUSSION TITLE
Add option for passwords to expire periodically

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5872] [Feature Requests]
> Ideally, we would expire them, and then upon their next successful authentication, they are forced to create a new password.
> I would like to do this in bulk for all users in the system.

**Requested By:** Matthew Cruz at The University of Michigan

**Comment by maxpatiiuk** [comment]
All modern cybersecurity experts advise against password expirations, in favor of encouraging your users to use 2fa, passkeys and password managers

https://www.techtarget.com/searchsecurity/tip/Why-its-time-to-expire-mandatory-password-expiration-policies?Offer=abMeterCharCount_var1

**Comment by maxpatiiuk** [comment]
https://cryptosmith.com/password-sanity/exp-harmful/
https://www.sans.org/blog/time-for-password-expiration-to-die/
https://www.pcmag.com/news/microsoft-realizes-password-expiration-is-poor-security

**Comment by ghost** [comment]
Obviously not a member of your dev team and so don't have a say on whether this ends up getting implemented or not, but if I can humbly ask if you do to at the very least please give us the ability to disable. I am in complete agreement with @maxpatiiuk and would not wish to implement this. For further direct guidance against this, one can consult section 10.2.1 Memorized Secrets from the [NIST digital identity guidelines](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-63b.pdf.)

Again, just my two cents, don't want to take over the conversation but I saw the issue and feel quite strongly about it.

**Comment by grantfitzsimmons** [comment]
@mpitblado This is simply a feature request from a user, so there will be much deliberation before any decision is made! In any case, something like this would never be required or done by default.

Thank you for leaving a comment! :smile:
